enhancements/ingress/configurable-dns-loadbalancerservice.md

```diff
@@ -0,0 +1,343 @@
+*DNSRecord* is left in an `Unmanaged` state which could result in orphaned resources
+on the cluster and the cloud provider.
+
+## Design Details
+
+### Open Questions
+
+Currently, as per documentation `.spec.domain` in the ingresscontroller cannot be
+updated after creation since updating the [domain](https://github.com/openshift/cluster-ingress-operator/blob/972f09b9dbb181ae5c414da2a990b57c60fde9d8/pkg/operator/controller/ingress/controller.go#L342-L355)
+in the ingresscontroller doesn't result in [updating](https://github.com/openshift/cluster-ingress-operator/blob/972f09b9dbb181ae5c414da2a990b57c60fde9d8/pkg/operator/controller/ingress/dns.go#L97)
+the wildcard *DNSRecord*. 
+> 1. Can this logic be changed to support moving from managed DNS to
+     unmanaged DNS on the `default` ingresscontroller?
+     
+  __Closed:__ The operator will continue to retain immutability of `.spec.domain`
+              and will not support updating domain when migrating from `Managed` to 
+              `Unmanaged` DNS.
+  
+
+### Test Plan
+
+- Test by creating a new secondary ingresscontroller with a custom domain
+  and `.loadbalancer.dnsManagementPolicy` set to `Unmanaged`. Ensure proper conditions
+  and connectivity.
+- Test deletion of ingresscontrollers to ensure correct behaviour of the associated
+  *DNSRecord* CR and the DNS record on the cloud provider.  
+- Test the following update paths on a custom ingresscontroller
+  - Updating `.loadbalancer.dnsManagementPolicy` from `Managed` -> `Unmanaged` -> `Managed` and
+    to ensure no conflicts during creation/recreation of DNS record on the cloud provider.
+  - Updating `Unmanaged` -> `Managed` and to ensure creation of new *DNSRecord*
+    and correct conditions.
+
+### Graduation Criteria
+
+N/A.
+
+#### Dev Preview -> Tech Preview
+
+N/A.  This feature will go directly to GA.
+
+#### Tech Preview -> GA
+
+N/A.  This feature will go directly to GA.
+
+#### Removing a deprecated feature
+
+N/A.
+
+### Upgrade / Downgrade Strategy
+
+On upgrades, `.loadbalancer.dnsManagementPolicy` will default to `Managed` which is consistent with the
+older versions of the ingress operator.
+
+On downgrades, there are 2 possible variations
+* When downgrading to an unsupported version, having the `.loadbalancer.dnsManagementPolicy` set to
+  `Unmanaged`,
+  * If the cluster domain is not the same as the domain in the ingresscontroller,
+    could result in failing to create the wildcard *DNSRecord* associated on the 
+    ingresscontroller and thereby result in the `cluster-ingress-operator` entering 
+    a degraded state. To recover DNS management will have to be disabled cluster 
+    wide ([docs](https://github.com/openshift/installer/blob/master/docs/user/aws/install_upi.md#remove-dns-zones)).
+  * When the cluster domain is the same as the domain in the ingresscontroller,
+    this will not cause any issues during downgrades, but a new wildcard *DNSRecord*
```

Comment:
If I understand correctly, the CR would already exist, but the DNSRecord CR's `spec.dnsManagementState` field would go away, and the downgraded ingress operator's dns controller would proceed to create the _actual_ DNS record.
```suggestion
    this will not cause any issues during downgrades, but a new wildcard DNS record
```

Reply:
Yes, that is what would happen.